lib/src/metta/runner/builtin_mods/json.rs

```diff
@@ -0,0 +1,223 @@
+    let result = space.borrow().subst(&Atom::expr([Atom::var("k"), Atom::var("v")]),
+                                      &Atom::expr([Atom::var("k"), Atom::var("v")]));
+    if result.len() == 0 {
+        return Err(ExecError::from("Input space must be a dictionary space which contains (key value) pairs. See dict-space function"));
+    }
+    let mut res_str = "".to_string();
+    for res_atom in result {
+        let res_expr = TryInto::<&ExpressionAtom>::try_into(&res_atom)?;
+        let key = match Str::from_atom(res_expr.children().get(0).unwrap()).ok_or_else(|| ExecError::from("Key must be a string")){
+            Ok(str) => str.to_string(),
+            Err(err) => return Err(err),
+        };
+        let value = res_expr.children().get(1).unwrap();
+        let value_str = encode_atom(value)?;
+        res_str = res_str + ", " + &key + ": " + &value_str;
+    }
+    res_str.remove(0);
+    res_str.remove(0);
+    res_str = format!("{{{}}}", res_str);
+    Ok(res_str)
+}
+
+fn encode_atom(input: &Atom) -> Result<String, ExecError> {
+    let metatype = atom_to_string(&get_meta_type(&input));
+    let space = DynSpace::new(GroundingSpace::new());
+    let typ = get_atom_types(&space, &input);
+    if metatype == "Expression" {
+        let exp = TryInto::<&ExpressionAtom>::try_into(input)?;
+        match encode_list(exp.clone().into_children()) {
+            Ok(encoded_list) => Ok(encoded_list),
+            Err(err) => Err(err),
+        }
+    }
+    else if atom_to_string(&typ[0]) == "Number" {
+        let number_atom_string = atom_to_string(&input);
+        Ok(number_atom_string)
+    }
+    else if atom_to_string(&typ[0]) == "SpaceType" {
+        match encode_dictspace(&input) {
+            Ok(str) => Ok(str),
+            Err(err) => Err(err)
+        }
+    }
+    // String should be encoded using serde_json since it will encode "a" with additional escape
+    // characters. Symbols like just 'a' are not supported by serde_json so proposition is to encode
+    // them like strings.
+    else {
+        let atom_string = atom_to_string(&input);
+        match serde_json::to_string(&atom_string) {
+            Ok(encoded) => Ok(encoded),
+            Err(err) => Err(ExecError::from(format!("Encode string failed: {}", err))),
+        }
+    }
+}
+
+fn json_encode(args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+    let arg_error = || ExecError::from("json-encode expects atom to encode as input");
+    let input = args.get(0).ok_or_else(arg_error)?;
+
+    match encode_atom(input) {
+        Ok(str) => Ok(vec![Atom::gnd(Str::from_string(str))]),
+        Err(err) => Err(err)
+    }
+}
+
+fn decode_value(v: &Value) -> Result<Atom, ExecError> {
+    match v {
+        Value::Array(_) => {match decode_array(v.as_array().unwrap()) {
+            Ok(atom) => Ok(atom),
+            Err(err) => Err(err),
+        }},
+        Value::Null => Ok(Atom::gnd(Str::from_str("null"))),
```

Comment:
I would suggest using `Null` or `null` symbol instead as basically `null` in JSON means no value. String is not distinguishable from `"null"` string as a value.